data_steward/cdr_cleaner/manual_cleaning_rules/remediate_basics.py

```diff
@@ -0,0 +1,395 @@
+"""
+This script updates OBSERVATION, SURVEY_CONDUCT, and PERSON tables with the
+given new set of the basics survey responses.
+
+This is a one-off cleaning rule for DC-3016 hot-fix. It is not expected to
+run for every CDR release.
+
+Original Issues: DC-3016 and its subtasks.
+"""
+
+#Python imports
+import logging
+
+# Project imports
+from common import (JINJA_ENV, OBSERVATION, PERSON, SURVEY_CONDUCT)
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.cdr_cleaner.clean_cdr import (COMBINED,
+                                             CONTROLLED_TIER_DEID_BASE,
+                                             CONTROLLED_TIER_DEID_CLEAN, QUERY,
+                                             REGISTERED_TIER_DEID_BASE,
+                                             REGISTERED_TIER_DEID_CLEAN)
+from resources import ext_table_for, mapping_table_for
+from retraction.retract_utils import is_combined_dataset, is_deid_dataset
+
+LOGGER = logging.getLogger(__name__)
+
+ISSUE_NUMBERS = ['DC3016']
+
+CREATE_NEW_OBS_ID_LOOKUP = JINJA_ENV.from_string("""
+CREATE TABLE IF NOT EXISTS `{{project}}.{{sandbox_dataset}}.{{new_obs_id_lookup}}`
+(
+    source_observation_id INT64 NOT NULL OPTIONS(description="observation_id from the source table."),
+    dataset_id STRING NOT NULL OPTIONS(description="Dataset ID of the destination table."),
+    observation_id INT64 NOT NULL OPTIONS(description="Newly assigned observation_id for the destination dataset.")
+)
+OPTIONS
+(
+    description="Lookup table for observation_ids. New observation_ids are assigned when loaded so no observation_id will be a duplicate."
+)
+""")
+
+# NOTE In case of re-running remediation for the same dataset, you need to run
```

Comment:
We need the generated IDs to be stable across ALL of the pipeline datasets for release tags 9 and 10.  The ID generation uses `ROWNUMBER() OVER(ORDER BY observation_id)` which will provide different results in tables with differing numbers of rows. Because of this, 

1. Logic needs to be added to only generate this lookup table for the largest number of observations dataset (which is `2022q4r7_combined_release with AIAN`)
2. Subsequent runs of this cleaning rule on other datasets should not delete or recreate this table.
3. The order of running the phases should start with `2022q4r7_combined_release`
@hiro-mishima

Reply:
Thanks! Your point makes sense. I added the parameter `dataset_with_largest_observation_id ` so we can specify the dataset you mentioned. 
https://github.com/all-of-us/curation/pull/1489/commits/4b12430e03556fc106b1435f871ffd2f94aaa35d#diff-638190b97f9a117d7f9c2a0fc16996cb1608d05f353b383919366a8b7aec007fR50

I also added this `if` statement in `setup_rule` so this table creation and population occurs only once:
https://github.com/all-of-us/curation/pull/1489/commits/4b12430e03556fc106b1435f871ffd2f94aaa35d#diff-638190b97f9a117d7f9c2a0fc16996cb1608d05f353b383919366a8b7aec007fR345

Reply:
Thanks again, @hiro-mishima for your quick responses on this.  The `dataset_with_largest_observation_id` parameter and `if` statement are good, but I think the sandbox parameter value will change based on the `tag` for the run.  So, a different lookup table would be created for revision 9 and revision 10, when they should really use the same table.

Reply:
@molly-giddings @lrwb-aou 
Can r9 and r10 have different generated observation_ids? The current logic creates `_observation_id_map` for r9 and r10 individually. The reason behind it is to have `_observation_id_map` has only the necessary records for each release. If r9 and r10's observation_ids must be consistent, I will need to update both the script and the QC notebook. It shouldn't be too difficult. Let me know if I should go ahead and update the logic.

Reply:
Krishna helped me fix a few more lines: https://github.com/all-of-us/curation/commit/92f4c7bd84d03faa5d0e273723076ba51025c891

Reply:
@lrwb-aou and I reviewed and it looks good.  Thanks, @hiro-mishima & @ksdkalluri!